### ISSUE TITLE
Spells are incorrectly displayed on the stack

### ISSUE THREAD

**Issue by MorihiroS** [Card-Forge/forge#5634]
**Describe the bug**
Some spells have incorrect text when they're displayed in the stack:

- [ ] Assassin's Trophy (It says that a player who cast the spell (not the one who had their permanent broken!!) is looking for the "Basic.Land" card);
- [ ] All is Dust (It says "Sacrifice permanents"");
- [ ] Recross thе Paths (It says "NICKNAMEreveals...", without space before "reveals");
- [ ] Surgical Extraction (It says "CARDNAME (XX) until end of turn. Search...")

**To Reproduce**
1. AI/Player casts Assassin's Trophy/All is Dust/Recross thе Paths/Surgical Extraction
2. See wrong figure

**Expected behavior**
Assassin's Trophy (It should say that a player who had their permanent broken may looking for the basic land card.);
All is Dust (It should say "Sacrifice all colored permanents" or "Each player sacrifices all permanents they control that are one or more colors");
Recross thе Paths (It should say "NICKNAME reveals...");
Surgical Extraction (It should say like "NICKNAME choose target CARDNAME (XX) in a graveyard. Search..." )

**Additional context**
Similar with #4318

**Desktop**
 - OS: Windows 10
 - Java Version: 1.8.0_202-b08
 - Forge Version: 1.6.64-SNAPSHOT-07-13

**Screenshots**

![trophy](https://github.com/Card-Forge/forge/assets/115870280/96a0cc1c-f4d7-4eb5-bfd9-0d07b33e9cf9)
![all is dust](https://github.com/user-attachments/assets/45ea782b-b4c9-4ea4-ba2e-467741ab83d2)
![recr1](https://github.com/user-attachments/assets/f54f7720-f081-411b-9cd2-4f91b48385a0)
![surg](https://github.com/user-attachments/assets/394ca4fe-df1e-49c6-8e67-c0eb2ed4b698)

**Comment by MorihiroS** [comment]
Also problem with activated ability of Nahiri, the Harbinger:

![Nahiri](https://github.com/user-attachments/assets/85ab9c83-09dd-4eb3-b0c0-e0ac7e3131c6)

**Comment by Hanmac** [comment]
some of them can be fixed with using `ChangeTypeDesc`

**Comment by MorihiroS** [comment]
Add Tear Asunder and Sleight of Hand

**Comment by MorihiroS** [comment]
@dracontes, is this issue a part of what you are trying to fix in #6218, #6219 and #6220 ?

**Comment by dracontes** [comment]
> @dracontes, is this issue a part of what you are trying to fix in #6218, #6219 and #6220 ?

I don't think so. The only thing I'm attempting to address in this ball park on those PRs is how additional costs for permanent spells are displayed in different places in-game. From your issue list, I do believe I attempted to tackle Assassin's Trophy, only to realize that, at the time, it was a bit beyond me.

**Comment by Hanmac** [comment]
@dracontes you might look into the ones that use `Basic.Land` in their Valid, to add `ChangeTypeDesc`

@tool4ever, `ChangeTypeDesc` probably needs to be added to descriptiveKeys? So changeText Effects can use it better?
https://github.com/Card-Forge/forge/blob/6d0bf1924248a8188dcec04c2f5382cf7fa45700/forge-game/src/main/java/forge/game/CardTraitBase.java#L62-L63
